docs/schema_registry_tutorial.rst

```diff
@@ -0,0 +1,505 @@
+|sr| registers this schema `Payments` to the subject `transactions-value`, and returns the schema id of `1` to the producer.
+The producer caches this schema to schema id mapping for subsequent message writes, so it only contacts |sr| on first schema write.
+When a consumer reads this data, it sees the Avro schema id of `1` and sends a schema request to |sr|.
+|sr| retrieves the schema associated to schema id `1`, and returns the schema to the consumer.
+The consumer caches this schema to schema id mapping for subsequent message reads, so it only contacts |sr| on first schema id read.
+
+
+Auto Schema Registration
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Additionally, by default, client applications automatically register new schemas.
+If they produce new messages to a new topic, then they will automatically try to register new schemas.
+This is very convenient in development environments.
+In production, we recommend that client applications do not automatically register new schemas.
+They can be done outside the client application to provide control over when schemas are registered with |sr-long| and how they evolve.
+
+Within the application, disable automatic schema registration by setting the configuration parameter `auto.register.schemas=false`, as shown in the examples below.
+
+.. sourcecode:: java
+
+   props.put(AbstractKafkaAvroSerDeConfig.AUTO_REGISTER_SCHEMAS, false);
+
+To manually register the schema outside of the application, send the schema to |sr| and associate it with a subject, in this case `transactions-value`.  It returns a schema id of `1`.
+
+.. sourcecode:: bash
+
+   $ curl -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" --data '{"schema": "{\"type\":\"record\",\"name\":\"Payment\",\"namespace\":\"io.confluent.examples.clients.basicavro\",\"fields\":[{\"name\":\"id\",\"type\":\"string\"},{\"name\":\"amount\",\"type\":\"double\"}]}"}' http://localhost:8081/subjects/transactions-value/versions
+   {"id":1}
+
+
+Schema Evolution and Compatibility
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Changing Schemas
+^^^^^^^^^^^^^^^^
+
+Up till now, you have seen the benefit of |sr-long| as being centralized schema management that enables client applications to register and retrieve globally unique schema ids.
+The main value, however, is in enabling schema evolution.
+Similar to how APIs evolve and need to be compatible for all applications that rely on old and new versions of the API, schemas also evolve and likewise need to be compatible for all applications that rely on old and new versions of the schema.
+This schema evolution is a natural behavior of how applications and data develop over time.
+
+|sr-long| allows for schema evolution and provides compatibility checks to ensure that the contract between producers and consumers is not broken.
+This is especially important in Kafka in which producers and consumers are decoupled applications, and organizational totally different teams.
```

Comment:
```suggestion
This is especially important in Kafka, because producers and consumers are decoupled applications that are often developed by teams in different organizations.
```